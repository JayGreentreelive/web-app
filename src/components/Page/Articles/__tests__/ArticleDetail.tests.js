import React from 'react'
import { MockedProvider } from '@apollo/react-testing'
import renderer from "react-test-renderer"
import { GET_ARTICLE_BY_TITLE } from './mocks'

import { ArticleDetail } from '..'

describe("ArticleDetail", () => {

    it("renders without crashing", () => {
        const tree = renderer.create(
            <MockedProvider mocks={GET_ARTICLE_BY_TITLE} addTypename={false}>
                <ArticleDetail match={{ params: { articleTitle: 'article-1' } }} />
            </MockedProvider>
        )

        expect(tree).toMatchSnapshot()
    })
})
