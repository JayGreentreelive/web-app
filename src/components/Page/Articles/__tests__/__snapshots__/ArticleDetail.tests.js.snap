// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleDetail passes back a null article 1`] = `
<div>
  <h3
    class="text-center text-danger"
  >
    There was an error loading this content. Try refreshing the page.
  </h3>
</div>
`;

exports[`ArticleDetail renders an article 1`] = `
<div>
  <div
    class="container-fluid my-6 max-width-800"
  >
    <div
      class="row align-items-center"
    >
      <div
        class="col-12 col-md-12 order-first px-3"
      />
      <div
        class="col-12 col-md text-center px-3"
      >
        <h2
          class="mb-1 text-dark"
        >
          Article 1
        </h2>
        <h6
          class="mt-1 article-subtitle font-weight-light"
        >
          READ THIS GUYS!
        </h6>
        <div
          class="embed-responsive embed-responsive-16by9 rounded my-4"
        >
          <div
            class="fill d-flex justify-content-center align-items-center"
          >
            <div
              class="spinner-border text-primary"
              role="status"
            >
              <span
                class="sr-only"
              >
                Loading...
              </span>
            </div>
          </div>
        </div>
        <div
          class="py-4 d-flex align-items-center"
        >
          <div
            class="embed-responsive embed-responsive-1by1 rounded-circle author-image mr-3"
          >
            <div
              class="fill d-flex justify-content-center align-items-center"
            >
              <div
                class="spinner-border text-primary"
                role="status"
              >
                <span
                  class="sr-only"
                >
                  Loading...
                </span>
              </div>
            </div>
          </div>
          <div
            class="text-left d-flex flex-column"
          >
            <p
              class="my-1 font-weight-bold text-dark"
            >
              Todd Mullins
            </p>
            <p
              class="my-1 mb-0 font-weight-light"
            >
              September 28, 2019  •  10 min
            </p>
          </div>
        </div>
        <div
          class="article-body my-3 font-weight-light pb-4 text-left"
        >
          <p>
            This is my really good article on how to be such a Godly man. Your life will be changed if you keep reading.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ArticleDetail renders an article without a read time 1`] = `
<div>
  <div
    class="container-fluid my-6 max-width-800"
  >
    <div
      class="row align-items-center"
    >
      <div
        class="col-12 col-md-12 order-first px-3"
      />
      <div
        class="col-12 col-md text-center px-3"
      >
        <h2
          class="mb-1 text-dark"
        >
          Article 1
        </h2>
        <h6
          class="mt-1 article-subtitle font-weight-light"
        >
          READ THIS GUYS!
        </h6>
        <div
          class="embed-responsive embed-responsive-16by9 rounded my-4"
        >
          <div
            class="fill d-flex justify-content-center align-items-center"
          >
            <div
              class="spinner-border text-primary"
              role="status"
            >
              <span
                class="sr-only"
              >
                Loading...
              </span>
            </div>
          </div>
        </div>
        <div
          class="py-4 d-flex align-items-center"
        >
          <div
            class="embed-responsive embed-responsive-1by1 rounded-circle author-image mr-3"
          >
            <div
              class="fill d-flex justify-content-center align-items-center"
            >
              <div
                class="spinner-border text-primary"
                role="status"
              >
                <span
                  class="sr-only"
                >
                  Loading...
                </span>
              </div>
            </div>
          </div>
          <div
            class="text-left d-flex flex-column"
          >
            <p
              class="my-1 font-weight-bold text-dark"
            >
              Todd Mullins
            </p>
            <p
              class="my-1 mb-0 font-weight-light"
            >
              September 28, 2019  •   min
            </p>
          </div>
        </div>
        <div
          class="article-body my-3 font-weight-light pb-4 text-left"
        >
          <p>
            This is my really good article on how to be such a Godly man. Your life will be changed if you keep reading.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ArticleDetail renders an article without a summary 1`] = `
<div>
  <div
    class="container-fluid my-6 max-width-800"
  >
    <div
      class="row align-items-center"
    >
      <div
        class="col-12 col-md-12 order-first px-3"
      />
      <div
        class="col-12 col-md text-center px-3"
      >
        <h2
          class="mb-1 text-dark"
        >
          Article 1
        </h2>
        <div
          class="embed-responsive embed-responsive-16by9 rounded my-4"
        >
          <div
            class="fill d-flex justify-content-center align-items-center"
          >
            <div
              class="spinner-border text-primary"
              role="status"
            >
              <span
                class="sr-only"
              >
                Loading...
              </span>
            </div>
          </div>
        </div>
        <div
          class="py-4 d-flex align-items-center"
        >
          <div
            class="embed-responsive embed-responsive-1by1 rounded-circle author-image mr-3"
          >
            <div
              class="fill d-flex justify-content-center align-items-center"
            >
              <div
                class="spinner-border text-primary"
                role="status"
              >
                <span
                  class="sr-only"
                >
                  Loading...
                </span>
              </div>
            </div>
          </div>
          <div
            class="text-left d-flex flex-column"
          >
            <p
              class="my-1 font-weight-bold text-dark"
            >
              Todd Mullins
            </p>
            <p
              class="my-1 mb-0 font-weight-light"
            >
              September 28, 2019  •  10 min
            </p>
          </div>
        </div>
        <div
          class="article-body my-3 font-weight-light pb-4 text-left"
        >
          <p>
            This is my really good article on how to be such a Godly man. Your life will be changed if you keep reading.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ArticleDetail renders an article without a title 1`] = `
<div>
  <div
    class="container-fluid my-6 max-width-800"
  >
    <div
      class="row align-items-center"
    >
      <div
        class="col-12 col-md-12 order-first px-3"
      />
      <div
        class="col-12 col-md text-center px-3"
      >
        <h6
          class="mt-1 article-subtitle font-weight-light"
        >
          READ THIS GUYS!
        </h6>
        <div
          class="embed-responsive embed-responsive-16by9 rounded my-4"
        >
          <div
            class="fill d-flex justify-content-center align-items-center"
          >
            <div
              class="spinner-border text-primary"
              role="status"
            >
              <span
                class="sr-only"
              >
                Loading...
              </span>
            </div>
          </div>
        </div>
        <div
          class="py-4 d-flex align-items-center"
        >
          <div
            class="embed-responsive embed-responsive-1by1 rounded-circle author-image mr-3"
          >
            <div
              class="fill d-flex justify-content-center align-items-center"
            >
              <div
                class="spinner-border text-primary"
                role="status"
              >
                <span
                  class="sr-only"
                >
                  Loading...
                </span>
              </div>
            </div>
          </div>
          <div
            class="text-left d-flex flex-column"
          >
            <p
              class="my-1 font-weight-bold text-dark"
            >
              Todd Mullins
            </p>
            <p
              class="my-1 mb-0 font-weight-light"
            >
              September 28, 2019  •  10 min
            </p>
          </div>
        </div>
        <div
          class="article-body my-3 font-weight-light pb-4 text-left"
        >
          <p>
            This is my really good article on how to be such a Godly man. Your life will be changed if you keep reading.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ArticleDetail renders an article without an image 1`] = `
<div>
  <div
    class="container-fluid my-6 max-width-800"
  >
    <div
      class="row align-items-center"
    >
      <div
        class="col-12 col-md-12 order-first px-3"
      />
      <div
        class="col-12 col-md text-center px-3"
      >
        <h2
          class="mb-1 text-dark"
        >
          Article 1
        </h2>
        <h6
          class="mt-1 article-subtitle font-weight-light"
        >
          READ THIS GUYS!
        </h6>
        <div
          class="py-4 d-flex align-items-center"
        >
          <div
            class="embed-responsive embed-responsive-1by1 rounded-circle author-image mr-3"
          >
            <div
              class="fill d-flex justify-content-center align-items-center"
            >
              <div
                class="spinner-border text-primary"
                role="status"
              >
                <span
                  class="sr-only"
                >
                  Loading...
                </span>
              </div>
            </div>
          </div>
          <div
            class="text-left d-flex flex-column"
          >
            <p
              class="my-1 font-weight-bold text-dark"
            >
              Todd Mullins
            </p>
            <p
              class="my-1 mb-0 font-weight-light"
            >
              September 28, 2019  •  10 min
            </p>
          </div>
        </div>
        <div
          class="article-body my-3 font-weight-light pb-4 text-left"
        >
          <p>
            This is my really good article on how to be such a Godly man. Your life will be changed if you keep reading.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ArticleDetail renders an article without author information 1`] = `
<div>
  <div
    class="container-fluid my-6 max-width-800"
  >
    <div
      class="row align-items-center"
    >
      <div
        class="col-12 col-md-12 order-first px-3"
      />
      <div
        class="col-12 col-md text-center px-3"
      >
        <h2
          class="mb-1 text-dark"
        >
          Article 1
        </h2>
        <h6
          class="mt-1 article-subtitle font-weight-light"
        >
          READ THIS GUYS!
        </h6>
        <div
          class="embed-responsive embed-responsive-16by9 rounded my-4"
        >
          <div
            class="fill d-flex justify-content-center align-items-center"
          >
            <div
              class="spinner-border text-primary"
              role="status"
            >
              <span
                class="sr-only"
              >
                Loading...
              </span>
            </div>
          </div>
        </div>
        <div
          class="py-4 d-flex align-items-center"
        >
          <div
            class="text-left d-flex flex-column"
          >
            <p
              class="my-1 font-weight-bold text-dark"
            >
              Todd Mullins
            </p>
            <p
              class="my-1 mb-0 font-weight-light"
            >
              September 28, 2019  •  10 min
            </p>
          </div>
        </div>
        <div
          class="article-body my-3 font-weight-light pb-4 text-left"
        >
          <p>
            This is my really good article on how to be such a Godly man. Your life will be changed if you keep reading.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ArticleDetail renders the error state 1`] = `
<div>
  <h3
    class="text-center text-danger"
  >
    There was an error loading this content. Try refreshing the page.
  </h3>
</div>
`;

exports[`ArticleDetail renders the loading state 1`] = `
<div>
  <div
    class="fill d-flex justify-content-center align-items-center"
  >
    <div
      class="spinner-border text-primary"
      role="status"
    >
      <span
        class="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
